crates/fj-kernel/src/algorithms/sweep.rs

```diff
@@ -124,11 +124,22 @@ pub fn sweep_shape(
                 let top_edge =
                     source_to_top.edges().get(&edge_source).unwrap().clone();
 
-                let surface =
+                let surface = if path.dot(&Vector::from([0., 0., 1.]))
```

Comment:
Would be nicer, if that check happened only once, and the result was stored into a variable with a descriptive name (`sweep_along_negative_direction`, or something like that).

With two separate checks, it becomes easy to miss one, once they need to change.